server/src/test/java/io/crate/expression/scalar/ArrayAppendFunctionTest.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.expression.scalar;
+
+import io.crate.expression.symbol.Literal;
+import io.crate.types.ArrayType;
+import io.crate.types.DataTypes;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static io.crate.testing.SymbolMatchers.isFunction;
+import static io.crate.testing.SymbolMatchers.isLiteral;
+
+public class ArrayAppendFunctionTest extends ScalarTestCase {
+
+    private static final ArrayType<Integer> INTEGER_ARRAY_TYPE = new ArrayType<>(DataTypes.INTEGER);
+
+    @Test
+    public void testNormalizeWithValueSymbols() throws Exception {
+        assertNormalize("array_append([10, 20], 30)", isLiteral(Arrays.asList(10, 20, 30)));
+    }
+
+    @Test
+    public void testNormalizeWithTextValueSymbols() throws Exception {
+        assertNormalize("array_append(['10', '20'], '30')", isLiteral(Arrays.asList("10", "20", "30")));
```

Comment:
Any reason to add an explicit string test, isn't is using the same code path like any other array inner type?

Reply:
No, not really. Was just checking by hand string values and added some test. 
Removed :)